lib/node_modules/@stdlib/math/base/napi/ternary/src/main.c

```diff
@@ -172,3 +172,79 @@ napi_value stdlib_math_base_napi_fff_f( napi_env env, napi_callback_info info, f
 
 	return v;
 }
+
+/**
+* Invokes a ternary function accepting a double-precision floating-point number and two signed 32-bit integers and returning a double-precision floating-point number.
+*
+* ## Notes
+*
+* -   This function expects that the callback `info` argument provides access to the following JavaScript arguments:
+*
+*     -   `x`: input value.
+*     -   `y`: input value.
+*     -   `z`: input value.
+*
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @param fcn    ternary function
+* @return       function return value as a Node-API double-precision floating-point number
+*/
+napi_value stdlib_math_base_napi_dii_d( napi_env env, napi_callback_info info, double (*fcn)( double, int32_t, int32_t ) ) {
+	napi_status status;
+
+	size_t argc = 3;
+	napi_value argv[ 3 ];
+	status = napi_get_cb_info( env, info, &argc, argv, NULL, NULL );
+	assert( status == napi_ok );
+
+	if ( argc < 3 ) {
+		status = napi_throw_error( env, NULL, "invalid invocation. Must provide three numbers." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype0;
+	status = napi_typeof( env, argv[ 0 ], &vtype0 );
+	assert( status == napi_ok );
+	if ( vtype0 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. First argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype1;
+	status = napi_typeof( env, argv[ 1 ], &vtype1 );
+	assert( status == napi_ok );
+	if ( vtype1 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. Second argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype2;
+	status = napi_typeof( env, argv[ 2 ], &vtype2 );
+	assert( status == napi_ok );
+	if ( vtype2 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. Third argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	double x;
+	status = napi_get_value_double( env, argv[ 0 ], &x );
+	assert( status == napi_ok );
+
+	int32_t y;
+	status = napi_get_value_int32_t( env, argv[ 1 ], &y );
```

Comment:
See https://nodejs.org/api/n-api.html#napi_get_value_int32

```suggestion
	status = napi_get_value_int32( env, argv[ 1 ], &y );
```